documentation/modules/ROOT/pages/configuration/topic-routing.adoc

```diff
@@ -7,46 +12,64 @@ include::../_attributes.adoc[]
 
-== Use-Cases
+The `ByLogicalTableRouter` transformation is a 
+link:https://kafka.apache.org/documentation/#connect_transforms[Kafka Connect SMT].
 
-=== Logical Tables
+ifdef::product[]
+The following topics provide details: 
 
-A logical table consists of one or more physical tables with the same table structure.
-A common use case is sharding, where for example two physical tables `db_shard1.my_table` and `db_shard2.my_table` together form one logical table.
+* xref:use-case-for-routing-records-to-topics-that-you-specify[]
+* xref:example-of-routing-records-for-multiple-tables-to-one-topic[]
+* xref:ensuring-unique-keys-across-records-routed-to-the-same-topic[]
+endif::product[]
 
-Typically the physical tables share the same schema.
+ifdef::community[]
+== Use case
+endif::community[]
 
-Normally, Debezium connectors send each change event to a topic that is named by the database and table.
-But since the sharded tables have the same schema, we'd instead like to re-route each change event to a topic named by the _logical_ table name.
-This way, all changes events for any of the shards all go to the same topic.
+ifdef::product[]
+// Type: concept
+[id="use-case-for-routing-records-to-topics-that-you-specify"]
+== Use case for routing records to topics that you specify
+endif::product[]
 
-What happens if each physical table has a primary key that is only unique within that table?
-In this case, a row in shard 1 can have the same primary key as a row in shard 2.
-Since Debezium events are keyed by the columns that make up the primary key, the events for that row in shard 1 would have the same key as the row in shard 2,
-even though globally they are different rows.
-So, in addition to changing the topic name, we may also want to _modify the event key_ to add a field that makes the key globally unique.
+The default behavior is that a Debezium connector sends each change event record to a topic whose name is formed from the name of the database and the name of the table in which the change was made. In other words, a topic receives records for one physical table. When you want a topic to receive records for more than one physical table, you must configure the Debezium connector to re-route the records to that topic. 
 
-This SMT lets you specify how you want to choose the new topic name and then specify how to modify the change event key to ensure it is globally unique.
+A logical table is a common use case for routing records for multiple physical tables to one topic. In a logical table, there are multiple physical tables that all have the same schema. For example, sharded tables have the same schema. A logical table might consist of two or more sharded tables: `db_shard1.my_table` and `db_shard2.my_table`. The tables are in different shards and are physically distinct but together they form a logical table. 
+You can re-route change event records for tables in any of the shards to the same topic.
 
-=== Partitioned Postgres Tables
```

Comment:
This content was not in the source file I worked on. I was pretty good about running `git pull upstream master`, but I missed it. Then when there were merge conflicts, Chris asked me to resolve them, and I had to make a choice between this content and my updated content. Looking back, I should have asked someone about this content since I had never seen it before. Instead, I chose my updated content thinking that perhaps this content was old. So certainly my error. In the future, I will ask before I resolve a conflict with content I have not seen before. 

How can I know when someone adds new content to a file that I am working in?
And since I am working on files in the master branch, how would I know if someone adds content that is only for, for example 1.2 when I am working toward a merge to 1.1?  I'm concerned about a situation where you add content for only 1.2 and I update that 1.2 content to fix a typo or clarify a sentence. Does my update to the 1.2 content get merged to 1.1 along with my other updates in that file?

Reply:
> How can I know when someone adds new content to a file that I am working in?

The most obvious is when you get merge conflicts, but if git is able to fast-forward, you could always check the commit history after your rebase to see what changes may have been recently applied that git was able to fast-forward.

> And since I am working on files in the master branch, how would I know if someone adds content that is only for, for example 1.2 when I am working toward a merge to 1.1?

I would say one of the easiest ways would be to review the file's git history and lookup the associated issue number.  If that's not enough, feel free to ask about any discrepency.

> I'm concerned about a situation where you add content for only 1.2 and I update that 1.2 content to fix a typo or clarify a sentence. Does my update to the 1.2 content get merged to 1.1 along with my other updates in that file?

No.

This process heavily depends on what changes need to be made and how those changes are to be applied to a given branch.  If those changes are 1.1 specific but can be applied in a version agnostic way, then a cherry-pick would suffice for 1.1 from master.  But if the changes are intertwined with version-specific features, you may end up creating multiple pull requests each designed to target a specific branch.

Reply:
Thanks very much for the info. Now I have some alternatives for what to do for a merge conflict. Let me know if there is anything I can do to get the missing postgresql partition content back.

Reply:
I think the best way might be to check-out master, make a copy topic-routing.adoc and then re-compare it with what you've done in your topic branch; adding back in what Gunnar mentioned was missing.  Once that's done I can give this a final review and merge.

Reply:
@Naros  - is this the content that you just referred to in chat? If yes, I don't know what I need to do.

Reply:
I just pushed a commit that returns the PostgreSQL partitioned table content to topic-routing.adoc.